File: src/main/java/br/com/rotaractsorocabaleste/rotasales/user/UserDetailsServiceImpl.java
package br.com.rotaractsorocabaleste.rotasales.user;

import lombok.RequiredArgsConstructor;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

import javax.transaction.Transactional;

import static java.lang.String.format;

@Service
@RequiredArgsConstructor
public class UserDetailsServiceImpl implements UserDetailsService {

    private final UserServiceImpl userServiceImpl;

    @Override
    @Transactional
    public UserDetails loadUserByUsername(final String username) throws UsernameNotFoundException {
        return new UserDetailsImpl(userServiceImpl.findByUsername(username).orElseThrow(() -> new UsernameNotFoundException(
                format("User: %s, not found", username)
        )));
    }

}

File: src/main/kotlin/br/com/rotaractsorocabaleste/rotasales/core/service/impl/UserDetailsServiceImpl.kt
package br.com.rotaractsorocabaleste.rotasales.core.service.impl

import br.com.rotaractsorocabaleste.rotasales.core.entity.UserDetailsImpl
import br.com.rotaractsorocabaleste.rotasales.core.service.UserService
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class UserDetailsServiceImpl(
    private val userService: UserService
) : UserDetailsService {

    @Transactional
    override fun loadUserByUsername(username: String?): UserDetails {
        return UserDetailsImpl(
            userService
                .findByUsername(username.toString())
                .first()
        )
    }

}